### DISCUSSION TITLE
velocity_decel_scale should be 1 or higher when input shaping is enabled

### DISCUSSION THREAD

**Discussion by ajshell1** [jschuh/klipper-macros#133] [General]
From Klipper's docs:

```
#max_accel_to_decel:
#   A pseudo acceleration (in mm/s^2) controlling how fast the
#   toolhead may go from acceleration to deceleration. It is used to
#   reduce the top speed of short zig-zag moves (and thus reduce
#   printer vibration from these moves). The default is half of
#   max_accel.
```

However, if you have input shaping enabled, input shaping handles the reduction of vibration itself, so accel_to_decel isn't needed, and will only slow you down. And Klipper ignores the accel_to_decel value when it is set to above max_accel. (This information has been sourced from several members of the Klipper Discord)

Unfortunately, klipper-macros' m204 gcode uses velocity_decel_scale to change the accel_to_decel, which is set to 0.5 by default. PrusaSlicer uses M204 even with their Klipper firmware support (added in 2.6.0 Alpha 6), and since I use their acceleration controls (originally just to increase accel for travel moves, now only to reduce accel on the first layer), I've ended up with a reduced accel_to_decel by accident.

Thus, I'd like at minimum a warning in the readme/the globals.cfg file about this. 

However, I think it would also be good if if the macros could detect if input shaping was enabled and automatically set the accel_to_decel based on that, although I'm not sure if that's possible.

**Comment by jschuh** [comment]
So, the problem is that it's easy to make a mistake where you've changed `max_accel_to_decel` in your `printer` config, but didn't realize you also had to change a related but different value in the macro config? That makes sense.

Honestly, I'm doubtful that the `M204` override even makes sense in Klipper these days. I found it useful for improving quality on external perimeters before input shaping, but at this point it seems unnecessary. So, I could just get rid of `variable_velocity_decel_scale` entirely, and remove the `M204` override along with the corresponding `M201` behavior.

Or, I suppose I could make it optional, and have that version infer a scale value from the `printer` config (I.e. set `ACCEL_TO_DECEL = ACCEL * max_accel_to_decel / max_accel`. I could even have it check for input shaping and infer whether to scale or use fixed limits based on that. But, I doubt that complexity is warranted.

I'm open to additional feedback. Although right now I'm leaning towards just removing it.

**Reply by jschuh** [reply]
FYI I posted a change to dev branch that just removes the `M204` override, etc.

**Comment by ajshell1** [comment]
Now that I think about it, I only use PrusaSlicer's acceleration control because I want a lower first layer acceleration, everything else is 2800mm/s^2.

Maybe I should just let it accelerate at that speed for the first layer. Not sure.
